component/distributed_cache/utils.go

```diff
@@ -187,3 +189,51 @@ func isMountPointRoot(path string) bool {
 	}
 	return false
 }
+
+// Get Dcache File size from the blob metadata property.
+func parseDcacheMetadata(attr *internal.ObjAttr) error {
+	log.Debug("DistributedCache::parseDcacheMetadata: file: %s", attr.Name)
+	var fileSize int64
+	var err error
+
+	if val, ok := attr.Metadata["cache-object-length"]; ok {
+		fileSize, err = strconv.ParseInt(*val, 10, 64)
+		if err == nil {
+			if fileSize >= 0 {
+				attr.Size = fileSize
+			} else if fileSize == -1 {
+				// FileSize can be negative in two cases:
+				// case1 : file is in writing state by the current node/ some other node in dcache. in which case we don't know
+				// the final size hence we return size to be zero.
+				// case2 : file got created and before the closing the fd of create, blobfuse got crashed. In that case we'll be
+				// having a stale entry which takes up the path and disallows the further file creations on that path.
+				// Now such files would be having the file size 0. but now when user tries to do the stat of this file they get
+				// the file size to be 0, Now how can they distiguish this file from the file which was actually of fileSize 0?
+				// Todo : is it a good idea to return file Size to be of INT64_MAX to distinguish these files?
```

Comment:
yes, listing such files with the special size is better than not listing them or listing with size 0.
so you can simply do

make sure fuse doesn't get confused with this. I want it to print the largest integer possible.